backend/monitoring/dashboard.py

```
@@ -707,6 +732,343 @@ async def trigger_test(test_type: str, admin=Depends(get_current_admin_dependenc
+            avatar_status["error"] = str(e)
+        
+        # Test MonetizationOptimizer
+        monetization_status = {"available": False, "error": None}
+        try:
+            optimizer = MonetizationOptimizer()
+            monetization_status["available"] = True
+        except Exception as e:
+            monetization_status["error"] = str(e)
+        
+        # Get recent spiritual service metrics
+        conn = await asyncpg.connect(DATABASE_URL)
```

Ruff (0.12.2)

Undefined name `asyncpg`

(F821)